src/compiler/types.ts

```diff
@@ -4068,31 +4087,40 @@ namespace ts {
         Tsx = ".tsx",
         Dts = ".d.ts",
         Js = ".js",
-        Jsx = ".jsx"
+        Jsx = ".jsx",
+        Json = ".json"
     }
 
     export interface ResolvedModuleWithFailedLookupLocations {
-        resolvedModule: ResolvedModuleFull | undefined;
+        readonly resolvedModule: ResolvedModuleFull | undefined;
         /* @internal */
-        failedLookupLocations: string[];
+        readonly failedLookupLocations: ReadonlyArray<string>;
+        /*@internal*/
+        isInvalidated?: boolean;

```

Comment:
This property doesn't really belong here -- this is the return type of functions in `moduleNameResolver.ts`, which have nothing to do with invalidation. Could you use a `{ resolved: ResolvedModuleWithFailedLookupLocations, isInvalidated: boolean }` tuple for your purposes instead?

Reply:
I dont think thats a good idea.. Already there are so many resolutions created for large projects, adding overhead of another object doesnt seem right. Instead I moved the invalidated definition to resolutionCache so shouldnt change anything in modulenameResolver